### PR TITLE
Fix missing clusters on user's cache

### PR DESCRIPTION
**Pull request by jlpadilla**
Signed-off-by: Jorge Padilla <jpadilla@redhat.com>

**Related Issue:**  https://issues.redhat.com/browse/ACM-2830

### Description of changes
- Add missing lock when reading managed clusters

### CODE REVIEW

File: pkg/rbac/userData.go [hunk 1]
```diff
@@ -340,15 +342,18 @@ func (user *UserDataCache) getSSRRforNamespace(ctx context.Context, cache *Cache
 func (user *UserDataCache) getNamespacedResources(cache *Cache, ctx context.Context,
 	clientToken string) (*UserDataCache, error) {
 	defer metric.SlowLog("UserDataCache::getNamespacedResources", 250*time.Millisecond)()
-	// check if we already have user's namespaced resources in userData cache and check if time is expired
+
+	// Lock the cache
 	user.nsrCache.lock.Lock()
 	defer user.nsrCache.lock.Unlock()
+	user.clustersCache.lock.Lock() // Lock the ManagedCluster cache because this function will update it.
+	defer user.clustersCache.lock.Unlock()
```
Comment (Anxhela21) [inline pkg/rbac/userData.go:350]: not sure we need these locks for clusterCache here

Comment (SherinV) [reply pkg/rbac/userData.go:350]: +1

Comment (SherinV) [reply pkg/rbac/userData.go]: FYI @jlpadilla I pushed the change to remove it. Will add it back if needed.

Comment (jlpadilla) [reply pkg/rbac/userData.go]: Thanks for debugging and fixing!

**Review by Anxhela21 — APPROVED** [review]
/lgtm

**Pull request merged**